src/devices/sound/lynx.cpp

```diff
@@ -233,35 +244,33 @@ void lynx_sound_device::count_down(int nr)
 
 void lynx_sound_device::shift(int chan_nr)
 {
-	int16_t out_temp;
+	s16 out_temp;
 	LYNX_AUDIO *channel;
 
 	assert(chan_nr < 4);
 
 	channel = &m_audio[chan_nr];
-	//channel->shifter = ((channel->shifter<<1)&0xffe) | (m_shift_xor[ channel->shifter & channel->mask ]&1);
+	//channel->shifter = ((channel->shifter << 1) & 0xffe) | (m_shift_xor[channel->shifter & channel->mask] & 1);
 
 	// alternative method (functionally the same as above)
-	uint8_t xor_out = 0;
+	u8 xor_out = 1; // output of xor is inverted
 	for (int bit = 0; bit < 12; bit++)
 	{
-		if ((channel->mask >> bit) & 1)
-			xor_out ^= (channel->shifter >> bit) & 1;
+		if (BIT(channel->mask, bit))
+			xor_out ^= BIT(channel->shifter, bit);
 	}
-	channel->shifter = ((channel->shifter << 1) & 0xffe) | (xor_out ^ 1); // output of xor is inverted
+	channel->shifter = ((channel->shifter << 1) & 0xffe) | xor_out;
 
 
-	if (channel->reg.control1 & 0x20) // integrate mode enabled
+	if (channel->reg.integrate_mode()) // integrate mode enabled
 	{
 		if (channel->shifter & 1)
 			out_temp = channel->reg.output + channel->reg.volume;
 		else
 			out_temp = channel->reg.output - channel->reg.volume;
 
 		// clipping
-		if (out_temp > 127) out_temp = 127;
-		if (out_temp < -128) out_temp = -128;
-		channel->reg.output = (int16_t)out_temp;
+		channel->reg.output = std::clamp<s16>((s16)out_temp, -128, 127);
```

Comment:
`out_temp` is already `s16` – there’s no need to cast it, or to explicitly specify the type template argument for `std::clamp`.